test/metabase/test/data/duckdb.clj

```diff
@@ -1,7 +1,109 @@
 (ns metabase.test.data.duckdb
-  (:require 
-  [metabase.test.data.sql-jdbc :as sql-jdbc.tx]))
+  (:require
+   [clojure.edn :as edn]
+   [clojure.java.io :as io]
+   [clojure.string :as str]
+   [metabase.config :as config]
+   [metabase.driver :as driver]
+   [metabase.test.data.interface :as tx]
+   [metabase.test.data.sql :as sql.tx]
+   [metabase.test.data.sql-jdbc :as sql-jdbc.tx]
+   [metabase.test.data.sql-jdbc.execute :as execute]
+   [metabase.test.data.sql-jdbc.load-data :as load-data]
+   [metabase.test.data.sql.ddl :as ddl]
+   [metabase.util :as u]
+   [metabase.util.log :as log]))
 
 (set! *warn-on-reflection* true)
 
-(sql-jdbc.tx/add-test-extensions! :duckdb)
\ No newline at end of file
+(sql-jdbc.tx/add-test-extensions! :duckdb)
+
+(doseq [[feature supported?] {:foreign-keys  (not config/is-test?) 
+                              :upload-with-auto-pk (not config/is-test?)}]
+  (defmethod driver/database-supports? [:duckdb feature] [_driver _feature _db] supported?))
+
+(defmethod tx/supports-time-type? :duckdb [_driver] false)
+
+(defmethod tx/dbdef->connection-details :duckdb [_ _ {:keys [database-name]}] 
+  {:read_only       false
+   :old_implicit_casting   true
+   "temp_directory"        (format "%s.ddb.tmp" database-name)
+   :database_file (format "%s.ddb" database-name)
+   "custom_user_agent"     "metabase_test"
+   :allow_unsigned_extensions  false
+   :subname                (format "%s.ddb" database-name)})
+
+(doseq [[base-type db-type] {:type/BigInteger     "BIGINT"
+                             :type/Boolean        "BOOL"
+                             :type/Date           "DATE"
+                             :type/DateTime       "TIMESTAMP"
+                             :type/DateTimeWithTZ "TIMESTAMPTZ"
+                             :type/Decimal        "DECIMAL"
+                             :type/Float          "FLOAT"
+                             :type/Integer        "INTEGER"
+                             :type/Text           "STRING"
+                             :type/Time           "TIME"
+                             :type/UUID           "UUID"}]
+  (defmethod sql.tx/field-base-type->sql-type [:duckdb base-type] [_ _] db-type))
+
+
+(defmethod sql.tx/pk-sql-type :duckdb [_] "INTEGER")
+
+(defmethod sql.tx/drop-db-if-exists-sql    :duckdb [& _] nil)
+(defmethod ddl/drop-db-ddl-statements   :duckdb [& _] nil)
+(defmethod sql.tx/create-db-sql         :duckdb [& _] nil)
+
+(defmethod tx/destroy-db! :duckdb
+  [_driver dbdef]
+  (let [file (io/file (str (tx/escaped-database-name dbdef) ".ddb"))
+        wal-file (io/file (str (tx/escaped-database-name dbdef) ".ddb.wal"))]
```

Comment:
duckdb.close() method is not called? That should clean up the WAL

Reply:
hmm, looking at the code, it's not apparent how connection is closed. and i don't see any .wal file locally any more. I'm guessing i was seeing them because some of the tests errored out and did not clean up connections.

Reply:
yeah, that would do it. Safer to keep the deletion then.